lib/membrane/realtimer.ex

```diff
@@ -39,7 +44,24 @@ defmodule Membrane.Realtimer do
         tick_actions: [buffer: {:output, buffer}] ++ state.tick_actions
     }
 
-    {[timer_interval: {:timer, interval}], state}
+    {maybe_start_timer ++ [timer_interval: {:timer, interval}], state}
+  end
+
+  @impl true
+  def handle_event(:input, %Events.Reset{}, _ctx, state) do
+    {actions, state} =
+      case state.tick_actions do
+        [] when state.timer_status == :to_be_started ->
+          {[], state}
+
+        [] when state.timer_status == :running ->
+          {[stop_timer: :timer], %{state | timer_status: :to_be_started}}
+
+        _many ->
+          {[], %{state | timer_status: :to_be_restarted}}
```

Comment:
Let's call that state `to_be_stopped` - it more appropriately reflects what that state indicates 
```suggestion
          {[], %{state | timer_status: :to_be_stopped}}
```